src/s3/async_multipart_put_object.rs

```diff
@@ -157,6 +157,101 @@ impl<'a> AsyncMultipartUpload<'a> {
         })
     }
 
+    /// Import an existing [AsyncMultipartUpload] to continue uploading.
+    ///
+    /// * `client`              - S3 client to use.
+    /// * `upload_id`           - The multipart upload ID to resume.
+    /// * `dst`                 - The [S3Object] to write the object into.
+    /// * `part_size`           - How large, in bytes, each part should be. Must be
+    /// larger than 5MIB and smaller that 5GIB.
+    /// * `max_uploading_parts` - How many parts to upload concurrently,
+    /// Must be larger than 0 (defaults to 100).
+    #[instrument(skip(client))]
+    pub async fn from(
+        client: &'a Client,
+        upload_id: String,
+        dst: &S3Object,
+        part_size: usize,
+        max_uploading_parts: Option<usize>,
+    ) -> anyhow::Result<AsyncMultipartUpload<'a>> {
+        event!(Level::DEBUG, "New AsyncMultipartUpload");
+        if part_size < MIN_PART_SIZE {
+            anyhow::bail!("part_size was {part_size}, can not be less than {MIN_PART_SIZE}")
+        }
+        if part_size > MAX_PART_SIZE {
+            anyhow::bail!("part_size was {part_size}, can not be more than {MAX_PART_SIZE}")
+        }
+
+        //Check that user did not send in invalid parameter
+        if let Some(0) = max_uploading_parts {
+            anyhow::bail!("Max uploading parts must not be 0")
+        }
+
+        let list_parts_result = client
+            .list_parts()
+            .bucket(&dst.bucket)
+            .key(&dst.key)
+            .max_parts(
+                max_uploading_parts
+                    .unwrap_or(DEFAULT_MAX_UPLOADING_PARTS)
```

Comment:
`DEFAULT_MAX_UPLOADING_PARTS` is used to control the maximum number of concurrent uploading chunks, where as in this API it is controlling the the maximum number of parts to return [link](https://docs.rs/aws-sdk-s3/latest/aws_sdk_s3/client/fluent_builders/struct.ListParts.html#method.set_max_parts).

I think this should either be left unset and accept this could return a large number of `Parts` or create a new constant and handle the pagination using the `part_number_marker` to fetch each set of `Parts` until the end is reached.